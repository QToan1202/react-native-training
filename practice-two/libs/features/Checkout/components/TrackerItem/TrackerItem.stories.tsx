import { ComponentMeta, ComponentStory } from '@storybook/react-native'

import { TrackerItem } from '../index'

export default {
  title: 'components/Tracker Item',
  component: TrackerItem,
} as ComponentMeta<typeof TrackerItem>

const Template: ComponentStory<typeof TrackerItem> = (args) => <TrackerItem {...args} />

export const Default = Template.bind({})
Default.args = {
  trackStatus: 'order placed',
  description: 'Order#123455 from Fashion Point',
  date: new Date(),
  time: '11:30 AM',
}
