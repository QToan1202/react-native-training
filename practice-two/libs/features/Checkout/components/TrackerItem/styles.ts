import { StyleSheet } from 'react-native'

import { COLORS, FONT_FAMILY } from '@practice-two/shared'

const styles = StyleSheet.create({
  container: {
    rowGap: 6,
  },
  text: {
    fontFamily: FONT_FAMILY.MONTSERRAT[500],
    color: COLORS.GRAY_400,
    fontSize: 10,
    lineHeight: 18,
  },
  title: {
    color: COLORS.DARK_50,
    fontSize: 14,
    textTransform: 'capitalize',
  },
})

export default styles
