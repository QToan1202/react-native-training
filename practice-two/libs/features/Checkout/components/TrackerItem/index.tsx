import { memo } from 'react'
import isEqual from 'react-fast-compare'
import { View, ViewProps } from 'react-native'

import Paragraph from '@components/Paragraph'
import { TTrackerStatus } from '@types'
import { containerStyles } from '@styles'
import styles from './styles'

export interface TrackerItemProps extends ViewProps {
  trackStatus: TTrackerStatus
  description: string
  date: Date
  time: string
}

const TrackerItem = ({
  trackStatus,
  description,
  date,
  time,
  style,
  ...rest
}: TrackerItemProps) => {
  return (
    <View style={[styles.container, style]} {...rest}>
      <View style={[containerStyles.inline, containerStyles.spaceBetween]}>
        <Paragraph content={trackStatus} style={[styles.text, styles.title]} />
        <Paragraph content={date.toLocaleDateString('en-GB')} style={styles.text} />
      </View>
      <View style={[containerStyles.inline, containerStyles.spaceBetween]}>
        <Paragraph content={description} style={styles.text} />
        <Paragraph content={time} style={styles.text} />
      </View>
    </View>
  )
}

export default memo(TrackerItem, isEqual)
