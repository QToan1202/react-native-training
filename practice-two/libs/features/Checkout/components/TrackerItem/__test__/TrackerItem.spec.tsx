import { render, screen } from '@testing-library/react-native'

import TrackerItem from '../index'

const mockedDate = new Date(2023, 7, 28)
jest.useFakeTimers()
jest.setSystemTime(mockedDate)

describe('Testing Tracker item component', () => {
  beforeEach(() => {
    render(
      <TrackerItem
        trackStatus="order placed"
        description="Order#123455 from Fashion Point"
        date={new Date()}
        time="11:30 AM"
        testID="tracker-item"
      />
    )
  })

  it('Render Tracker item success', () => {
    const component = render(
      <TrackerItem
        trackStatus="order placed"
        description="Order#123455 from Fashion Point"
        date={new Date()}
        time="11:30 AM"
      />
    ).toJSON()

    expect(component).toMatchSnapshot()
  })

  it('An item must have the title and description about status', () => {
    const component = screen.getByTestId('tracker-item')
    const title = screen.getByText('order placed')
    const description = screen.getByText('Order#123455 from Fashion Point')

    expect(component).toContainElement(title)
    expect(component).toContainElement(description)
  })
})
