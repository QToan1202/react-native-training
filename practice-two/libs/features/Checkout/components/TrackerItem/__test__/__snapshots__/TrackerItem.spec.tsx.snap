// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Tracker item component Render Tracker item success 1`] = `
<View
  style={
    [
      {
        "rowGap": 6,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "columnGap": 4,
          "flexDirection": "row",
        },
        {
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontFamily": "Montserrat_400Regular",
          },
          {
            "fontSize": 14,
          },
          [
            {
              "color": "#606a7b",
              "fontFamily": "Montserrat_500Medium",
              "fontSize": 10,
              "lineHeight": 18,
            },
            {
              "color": "#212121",
              "fontSize": 14,
              "textTransform": "capitalize",
            },
          ],
        ]
      }
    >
      order placed
    </Text>
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontFamily": "Montserrat_400Regular",
          },
          {
            "fontSize": 14,
          },
          {
            "color": "#606a7b",
            "fontFamily": "Montserrat_500Medium",
            "fontSize": 10,
            "lineHeight": 18,
          },
        ]
      }
    >
      28/08/2023
    </Text>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "columnGap": 4,
          "flexDirection": "row",
        },
        {
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontFamily": "Montserrat_400Regular",
          },
          {
            "fontSize": 14,
          },
          {
            "color": "#606a7b",
            "fontFamily": "Montserrat_500Medium",
            "fontSize": 10,
            "lineHeight": 18,
          },
        ]
      }
    >
      Order#123455 from Fashion Point
    </Text>
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontFamily": "Montserrat_400Regular",
          },
          {
            "fontSize": 14,
          },
          {
            "color": "#606a7b",
            "fontFamily": "Montserrat_500Medium",
            "fontSize": 10,
            "lineHeight": 18,
          },
        ]
      }
    >
      11:30 AM
    </Text>
  </View>
</View>
`;
